### PR TITLE
FInetune support added in infer-base

### PR DESCRIPTION
**Pull request by dev-nid**
### Problem

QvacResponse was designed around inference (runJob): the addon streams text chunks via Output events, and when JobEnded fires, the response resolves with the accumulated output array. Stats are optional metadata attached at the end.
Finetuning doesn't fit this model:

- **During the job:** Training emits FinetuneProgress events with per-iteration metrics (loss, learning rate, etc.) — not text output. These events were unhandled and silently dropped, so consumers had no way to track training progress.
- **At job completion:** The JobEnded payload for a finetune carries the terminal result ({ op: 'finetune', status: 'completed' }), not inference stats. The existing code fed this into updateStats() — overwriting real training metrics with terminal metadata — and then resolved the promise with the (empty) output array, discarding the actual finetune result.

In short: inference produces text outputs and resolves with them; finetuning produces progress stats and resolves with a terminal status. The base layer had no way to distinguish between the two.

#### BaseInference._outputCallback
- Added a FinetuneProgress event branch that routes data.stats to response.updateStats(), streaming per-iteration training metrics to consumers.
- In the JobEnded branch, added a isFinetuneTerminal check (data.op === 'finetune' && typeof data.status === 'string') to separate the two flows:

    - Finetune jobs: skips updateStats() (avoids overwriting training metrics with terminal metadata) and calls response.ended(data) to resolve with the terminal result.
    - Regular jobs: unchanged — stats are updated and response.ended() resolves with accumulated outputs.

#### QvacResponse.ended
- Changed signature from ended() to ended(result = this.output) so callers can pass a custom result while defaulting to the accumulated output array for backward compatibility.
#### QvacResponse.onFinish
- Changed from () => callback(this.output) to (result) => callback(result) so the finish callback receives whatever ended() resolved with instead of always reading the output array.

### CODE REVIEW

File: packages/qvac-lib-infer-base/package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@qvac/infer-base",
-  "version": "0.2.2",
+  "version": "0.2.3",
```
Comment (gianni-cor) [inline packages/qvac-lib-infer-base/package.json]: maybe it is better to bump package when we open PR in the release branch.  SO better if we just merge this PR without bumping

**Review by gianni-cor — CHANGES_REQUESTED** [review]

File: packages/qvac-lib-infer-base/package.json (hunk 1 above)
Comment (dev-nid) [reply packages/qvac-lib-infer-base/package.json]: Got it

**Review by gianni-cor — APPROVED** [review]

**Review by donriddo — APPROVED** [review]
